lib/core/http/api_response.dart

```diff
@@ -0,0 +1,22 @@
+enum ApiStatus {
+  loading,
+  completed,
+  error,
+}
+
+class ApiResponse<T> {
+  ApiStatus? status;
```

Comment:
Seems status doesn't have to be optional

Reply:
done